internal/services/servicebus/servicebus_topic_resource.go

```diff
@@ -283,6 +283,27 @@ func resourceServiceBusTopicCreateUpdate(d *pluginsdk.ResourceData, meta interfa
 		return fmt.Errorf("retrieving ServiceBus Namespace %q (Resource Group %q): %+v", id.NamespaceName, id.ResourceGroupName, err)
 	}
 
+	isPremiumNamespacePartitioned := true
+	var sku namespaces.SkuName
+	if nsModel := resp.Model; nsModel != nil {
+		sku = nsModel.Sku.Name
+	}
+
+	if sbNamespaceModel := resp.Model; sbNamespaceModel != nil {
+		if sbNamespaceModel.Properties != nil &&
+			sbNamespaceModel.Properties.PremiumMessagingPartitions != nil && *sbNamespaceModel.Properties.PremiumMessagingPartitions == 1 {
+			isPremiumNamespacePartitioned = false
+		}
+	}
+
+	if sku == namespaces.SkuNamePremium {
+		if isPremiumNamespacePartitioned && !enablePartitioning {
+			return fmt.Errorf("non-partitioned entities are not allowed in partitioned namespace")
```

Comment:
It would be helpful to rephrase the error message into something actionable for the user
```suggestion
			return fmt.Errorf("topic must have `partitioning_enabled` set to `true` when the parent namespace is partitioned")
```

Reply:
This hasn't been updated and there is no response as to why?

Reply:
updated